src/components/RadioSwitchField/RadioSwitchField.tsx

```diff
@@ -0,0 +1,97 @@
+import FormControl from "@material-ui/core/FormControl";
+import FormControlLabel from "@material-ui/core/FormControlLabel";
+import Radio from "@material-ui/core/Radio";
+import RadioGroup from "@material-ui/core/RadioGroup";
+import { createStyles, withStyles, WithStyles } from "@material-ui/core/styles";
+import classNames from "classnames";
+import React from "react";
+
+const styles = createStyles({
+  formControl: {
+    padding: 0,
+    width: "100%"
+  },
+  formLabel: {
+    marginLeft: "-5px",
+    paddingBottom: "10px"
+  },
+  radioLabel: {
+    "& > span": {
+      padding: "10px 6px"
+    }
+  },
+  secondLabel: {
+    display: "block",
+    fontSize: "12px"
+  }
+});
+
+interface RadioSwitchFieldProps {
+  className?: string;
+  disabled?: boolean;
+  error?: boolean;
+  firstOptionLabel: React.ReactNode;
+  name?: string;
+  secondOptionLabel: React.ReactNode;
+  value?: boolean;
+  onChange: (event: React.ChangeEvent<any>) => void;
+}
+
+export const RadioSwitchField = withStyles(styles, {
+  name: "RadioSwitchField"
+})(
+  ({
+    className,
+    classes,
+    disabled,
+    error,
+    firstOptionLabel,
+    onChange,
+    name,
+    secondOptionLabel,
+    value
+  }: RadioSwitchFieldProps & WithStyles<typeof styles>) => {
+    const initialValue = value ? "visible" : "hidden";
```

Comment:
`visible`/`hidden` pair won't be always valid options, I can imagine situations where it would be `on`/`off` or `yes`/`no`